app/lib/menu/menu-builder.js

```diff
@@ -198,19 +199,33 @@ class MenuBuilder {
       }
     }));
 
-    this.appendReopenLastTab();
+    this.appendOpenRecent();
 
     return this;
   }
 
-  appendReopenLastTab() {
+  appendOpenRecent() {
     this.menu.append(new MenuItem({
-      label: 'Reopen Last File',
-      enabled: this.options.state.lastTab,
-      accelerator: 'CommandOrControl+Shift+T',
-      click: function() {
-        app.emit('menu:action', 'reopen-last-tab');
-      }
+      label: 'Open Recent',
+      enabled: true,
+      submenu: Menu.buildFromTemplate([
+        {
+          label: 'Reopen Last File',
+          enabled: this.options.state.lastTab,
+          accelerator: 'CommandOrControl+Shift+T',
+          click: function() {
+            app.emit('menu:action', 'reopen-last-tab');
+          }
+        },
+        getSeparatorTemplate(),
+        ...(this.options.state.closedTabs.reverse().map((tab) => {
```

Comment:
Note that `.reverse()` modifies the object in place.

Reply:
Good catch, I will adjust it to use a cloned array for that 👍